supervision/annotators/core.py

```diff
@@ -1496,3 +1496,138 @@ def annotate(
                 )
 
         return scene
+
+
+class PercentageBarAnnotator(BaseAnnotator):
+    """
+    A class for drawing percentage bars of confidence threshold on an image
+    using provided detections.
+    """
+
+    def __init__(
+        self,
+        height: int = 16,
+        width: int = 80,
+        color: Union[Color, ColorPalette] = ColorPalette.default(),
+        border_color: Color = Color.black(),
+        position: Position = Position.TOP_CENTER,
+        color_lookup: ColorLookup = ColorLookup.CLASS,
+    ):
+        """
+        Args:
+            height (int): The height in pixels of the percentage bar.
+            width (int): The width in pixels of the percentage bar.
+            color (Union[Color, ColorPalette]): The color or color palette to use for
+                annotating detections.
+            border_color (Color): The color of the border lines.
+            position (Position): The anchor position of drawing the percentage bar.
+            color_lookup (str): Strategy for mapping colors to annotations.
+                Options are `INDEX`, `CLASS`, `TRACK`.
+        """
+        self.height: int = height
+        self.width: int = width
+        self.color: Union[Color, ColorPalette] = color
+        self.border_color: Color = border_color
+        self.position: Position = position
+        self.color_lookup: ColorLookup = color_lookup
+
+    def annotate(
```

Comment:
One thing we really should add to make it a killer annotator is allowing users to pass a custom NumPy array with values between 0 and 1. This way users would be able to plot confidence by default, but also any other percentage value.

Reply:
I made custom_value acceptability instead of using conf values and I added couple of different types and make sure values are between 0 and 1 as well. Please check it out.